presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusConnectorConfig.java

```diff
@@ -0,0 +1,128 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.ConfigurationException;
+import com.google.inject.spi.Message;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class PrometheusConnectorConfig
+{
+    private URI prometheusURI = new URI("http://localhost:9090");
+    private Duration queryChunkSizeDuration = new Duration(1, TimeUnit.DAYS);
+    private Duration maxQueryRangeDuration = new Duration(21, TimeUnit.DAYS);
+    private Duration cacheDuration = new Duration(30, TimeUnit.SECONDS);
+    private String bearerTokenFile;
+
+    public PrometheusConnectorConfig()
+            throws URISyntaxException
+    {}
+
+    @NotNull
+    public URI getPrometheusURI()
+    {
+        return prometheusURI;
+    }
+
+    @Config("prometheus-uri")
+    @ConfigDescription("Where to find Prometheus coordinator host")
+    public PrometheusConnectorConfig setPrometheusURI(URI prometheusURI)
+    {
+        this.prometheusURI = prometheusURI;
+        return this;
+    }
+
+    @MinDuration("1ms")
+    public Duration getQueryChunkSizeDuration()
+    {
+        return queryChunkSizeDuration;
+    }
+
+    public String getQueryChunkSizeDurationAsPrometheusCompatibleDurationString()
+    {
+        return String.valueOf(queryChunkSizeDuration.roundTo(queryChunkSizeDuration.getUnit())) +
+                io.airlift.units.Duration.timeUnitToString(queryChunkSizeDuration.getUnit());
+    }
+
+    @Config("query-chunk-size-duration")
+    @ConfigDescription("The duration of each query to Prometheus")
+    public PrometheusConnectorConfig setQueryChunkSizeDuration(String queryChunkSizeDuration)
+    {
+        this.queryChunkSizeDuration = Duration.valueOf(queryChunkSizeDuration);
+        return this;
+    }
+
+    @MinDuration("1ms")
+    public Duration getMaxQueryRangeDuration()
+    {
+        return maxQueryRangeDuration;
+    }
+
+    @Config("max-query-range-duration")
+    @ConfigDescription("Width of overall query to Prometheus, will be divided into query-chunk-size-duration queries")
+    public PrometheusConnectorConfig setMaxQueryRangeDuration(String maxQueryRangeDuration)
+    {
+        this.maxQueryRangeDuration = Duration.valueOf(maxQueryRangeDuration);
+        return this;
+    }
+
+    @MinDuration("1s")
+    public Duration getCacheDuration()
+    {
+        return cacheDuration;
+    }
+
+    @Config("cache.ttl")
```

Comment:
Please append `prometheus.` to each property. Also, how about renaming to `prometheus.table-name.cache-ttl` or something?

Reply:
Re-did naming of properties to be inline with other connectors in e84fda70
Added `prometheus.` everywhere. Changed `-` to `.` everywhere.